### PR TITLE
fix(resolve-template): don't split on quoted pipe characters

### PR DESCRIPTION
**Pull request by glepur**
https://github.com/nxtedition/nxt/issues/4129

### CODE REVIEW

File: src/util/template/expression.js [hunk 1]
```diff
@@ -335,8 +335,7 @@ module.exports = ({ ds } = {}) => {
 
   return memoize(expression => {
     try {
-      const [ basePath, ...tokens ] = expression.trim().split(/\s*\|\s*/)
-
+      const [ basePath, ...tokens ] = expression.trim().split(/\s*\|\s*(?=(?:(?:[^"]*"){2})*[^"]*$)/)
```
Comment (ronag) [inline src/util/template/expression.js]: When you need to do complicated things like this... someone else has probably already done it and there is a npm package. We are already using https://www.npmjs.com/package/split-string in this file. Why not use that?

https://github.com/nxtedition/nxt-lib/blob/master/src/util/template/expression.js#L309

Comment (glepur) [reply src/util/template/expression.js]: My intention was to use split-string, but it only takes string as delimiter, we split by `/\s*\|\s*/` .
I can try to find 3rd party lib that can solve the problem, but IMHO it's better to keep 3rd party dependencies as low as possible.

Comment (ronag) [reply src/util/template/expression.js]: @glepur split by '|' and the trim each token?

`split(str, '|').map(str => str.trim())`

Comment (glepur) [reply src/util/template/expression.js]: That would actually work! Awesome!
I'll do that!

**Comment by glepur** [comment]
Done.
Implemented your idea.

**Pull request merged**